### PR TITLE
Demo Version 0.0.1

### PR DESCRIPTION
**Pull request by rahulchauhan049**
This pull request contains the following changes
added a description,
added data,
demo shiny app
Update readme.md,
code format and minor issues fix like using the switch in place of if/else, use external CSS, a proper indentation of code

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Package: dashboard.demo
+Title: Dashboard For biodiversity data visualization
+Version: 0.0.1
+Authors@R: c(
+    person(
+        "Rahul", "Chauhan",
+        email = "rahul.chauhan049@gmail.com",
+        role = c("aut","cre")),
+    person(
```
Comment (thiloshon) [inline DESCRIPTION:9]: Rahul, Tomer, Vijay, Thiloshon Please

File: R/app_server.R [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#' @import shiny
+options(shiny.maxRequestSize = 5000 * 1024 ^ 2)
+
+app_server <- function(input, output, session) {
+  ##################Input tabs#################################
```
Comment (thiloshon) [inline R/app_server.R]: Remove ### comments, here its not needed.

File: R/app_server.R [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  
+  ##############################################################
+  
+  
+  
+  
+  
+  ##################Spatial tabs###############################
+  
+  callModule(mod_spatial_server, "spatial_ui_1", inputDataset)
```
Comment (thiloshon) [inline R/app_server.R]: remove '_1' in all ids. Just keep it as data_input, spatial_ui

File: R/mod_taxonomic.R [hunk 4]
```diff
@@ -0,0 +1,211 @@
+# Module UI
+
+#' @title   mod_taxonomic_ui and mod_taxonomic_server
+#' @description  A shiny Module.
```
Comment (thiloshon) [inline R/mod_taxonomic.R]: Add proper description

File: R/mod_taxonomic.R [hunk 5]
```diff
@@ -0,0 +1,211 @@
+# Module UI
```
Comment (thiloshon) [inline R/mod_taxonomic.R:1]: Add Proper title for this module

File: R/mod_taxonomic.R [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+    )
+  )
+}
+
+# Module Server
+
+#' @rdname mod_taxonomic
+#' @export
+#' @keywords internal
+
```
Comment (thiloshon) [inline R/mod_taxonomic.R:68]: remove space

File: R/mod_taxonomic.R [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+mod_taxonomic_server <- function(input, output, session, dataTaxo) {
```
Comment (thiloshon) [inline R/mod_taxonomic.R]: follow underscore standards. data_taxo. not dataTaxo

File: R/mod_taxonomic.R [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+    temp <- as.data.frame(table(data["genus"]))
+    data <- unique(data)
+    temp <- merge(data, temp , by.x = "genus", by.y = "Var1")
+    temp <- temp[c("phylum", "order", "family", "genus", "Freq")]
+    colnames(temp) <-
+      c("root", "group", "subgroup", "subsubgroup", "value")
+    
+    return(temp)
+  }
+
```
Comment (thiloshon) [inline R/mod_taxonomic.R]: remove extra lines

File: R/mod_taxonomic.R [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+    colnames(temp) <-
+      c("root", "group", "subgroup", "subsubgroup", "value")
+    
+    return(temp)
+  }
+  
+  
+}
+
+## To be copied in the UI
```
Comment (thiloshon) [inline R/mod_taxonomic.R]: remove golem instructions

File: R/mod_temporal.R [hunk 10]
```diff
@@ -0,0 +1,368 @@
...
+#' @keywords internal
+
+mod_temporal_server <-
+  function(input, output, session, dataTemporal) {
+    ns <- session$ns
+    formattedData <- reactive({
+      dataForBar <- format_bdvis(dataTemporal(), source = 'rgbif')
+      
+      
+      names(dataForBar) = gsub("\\.", "_", names(dataForBar))
```
Comment (thiloshon) [inline R/mod_temporal.R:94]: leave single space before

**Review by thiloshon — CHANGES_REQUESTED** [review]

File: DESCRIPTION (hunk 1 above)
Comment (rahulchauhan049) [reply DESCRIPTION:9]: Do you mean names order?

Comment (thiloshon) [reply DESCRIPTION:9]: Yes :)

File: DESCRIPTION [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+        role = c("aut"),
+        comment = c(ORCID = '0000-0003-1557-8596')),
+    person(
+        "Vijay", "Barve", ,
+        email = "vijay.barve@gmail.com", 
+        role = c("aut"),
+        comment = c(ORCID = '0000-0002-4852-2567'))
+    )
+Maintainer: Rahul Chauhan <rahul.chauhan049@gmail.com>
+Description: Provide feature to visualization biodiversity data. This is a easy to use shiny dashboard which allow to visualize different aspect of biodiversity data such as temporal, taxonomic, spatial.
```
Comment (thiloshon) [inline DESCRIPTION:25]: Make it 'Provides visualizations of biodiversity data. This is an easy to use shiny dashboard which allows visualizing different aspect of biodiversity like temporal, taxonomic and spatial.'

File: R/app_ui.R [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+    
+    #-------------------------Body Ends Here-------------------------------
+    
+  )#End of dashboard page
+}
+
+#' @import shiny
+#Function to attach Files such as css to shiny.
+golem_add_external_resources <- function() {
+  addResourcePath('www', system.file('app/www', package = 'bdvisShiny'))
```
Comment (thiloshon) [inline R/app_ui.R:100]: Change `package = 'bdvisShiny'` to `package = 'dashboard.demo'`

File: DESCRIPTION [hunk 13]
```diff
@@ -0,0 +1,49 @@
+Package: dashboard.demo
+Title: Dashboard For biodiversity data visualization
+Version: 0.0.1
```
Comment (thiloshon) [inline DESCRIPTION:3]: Add Date

File: .gitignore [hunk 14]
```diff
@@ -34,3 +34,8 @@ vignettes/*.pdf
```
Comment (thiloshon) [inline .gitignore]: Dont send gitignore. Rbuildignore should be sent, but not gitignore

File: R/app_ui.R [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+#' @import flexdashboard
+#' @import bdvis
+#' @import ggplot2
+#' @import tidyr
+#' @import treemap
+#' @import sunburstR
+app_ui <- function() {
+  dashboardPage(
+    skin = "green",
+    dashboardHeader(title = "bdvisashboard"),
```
Comment (thiloshon) [inline R/app_ui.R:19]: title = "bdvis Dashboard"

File: R/mod_dataInput.R [hunk 16]
```diff
@@ -0,0 +1,438 @@
...
+#' @param input internal
+#' @param output internal
+#' @param session internal
+#'
+#' @rdname mod_dataInput
+#'
+#' @keywords internal
+#' @export
+#' @importFrom shiny NS tagList
+mod_dataInput_ui <- function(id) {
```
Comment (thiloshon) [inline R/mod_dataInput.R:16]: Use the input module from bddwc.app as it is.

File: R/mod_dataSummary.R [hunk 17]
```diff
@@ -0,0 +1,655 @@
+# Module UI
```
Comment (thiloshon) [inline R/mod_dataSummary.R:1]: Add proper roxygen comments to all functions

File: R/mod_dataSummary.R [hunk 18]
```diff
@@ -0,0 +1,655 @@
...
+                  ns("species"),
+                  width = "100%"
+                )
+              )
+            )
+          )
+        )
+      )
+    )
+  )#End of fluidPage
```
Comment (thiloshon) [inline R/mod_dataSummary.R:161]: Remove comments like this #End of fluidPage.
What purpose does this serve?

File: R/mod_dataSummary.R [hunk 19]
```diff
@@ -0,0 +1,655 @@
...
+}
+
+# Module Server
+
+#' @rdname mod_dataSummary
+#' @export
+#' @keywords internal
+mod_dataSummary_server <-
+function(input, output, session, dataset) {
```
Comment (thiloshon) [inline R/mod_dataSummary.R:170]: Dont break as
mod_dataSummary_server <-
function 

keep it in same line

File: R/mod_dataSummary.R [hunk 20]
```diff
@@ -0,0 +1,655 @@
...
+    output$Gauge1 <- flexdashboard::renderGauge({
+      df <- dataset()
+      latitude <-
+        round(((nrow(df["decimalLatitude"]) - sum(is.na(
+          df["decimalLatitude"]
+        ))) / nrow(df["decimalLatitude"])), 2) * 100
+      longitude <-
+        round(((nrow(df["decimalLongitude"]) - sum(is.na(
+          df["decimalLongitude"]
+        ))) / nrow(df["decimalLongitude"])), 2) * 100
```
Comment (thiloshon) [inline R/mod_dataSummary.R:181]: Add space after this

File: R/mod_dataSummary.R [hunk 21]
```diff
@@ -0,0 +1,655 @@
...
+          nrow(df["countryCode"]),
+          nrow(df["locality"]))
+      
+      MissingRecords <-
+        c(sum(is.na(df["decimalLatitude"])),
+          sum(is.na(df["decimalLongitude"])),
+          sum(is.na(df["coordinateUncertaintyInMeters"])),
+          sum(is.na(df["coordinatePrecision"])),
+          sum(is.na(df["countryCode"])),
+          sum(is.na(df["locality"])))
```
Comment (thiloshon) [inline R/mod_dataSummary.R:287]: break after this please

File: R/mod_dataSummary.R [hunk 22]
```diff
@@ -0,0 +1,655 @@
...
+    output$temporalTable <- DT::renderDataTable({
+      df <- dataset()
+      names <- c("eventDate",
+                 "day",
+                 "month",
+                 "year",
+                 "dateIdentified",
+                 "lastInterpreted")
+      
+      TotalRecords <-
```
Comment (thiloshon) [inline R/mod_dataSummary.R:340]: Why Title case here? Use snake_case

File: R/mod_dataSummary.R [hunk 23]
```diff
@@ -0,0 +1,655 @@
...
+              nrow(df["lastInterpreted"])
+          ), 2) * 100, "%")
+        )
+      
+      tableTemporal <-
+        data.frame(names, TotalRecords, MissingRecords, RecordsPercentage)
+      tableTemporal
+    })
+    
+    output$box_A <-
```
Comment (thiloshon) [inline R/mod_dataSummary.R:392]: Dont use box_A cases. use snake_case. I was very specific about hadleys guide.

File: R/mod_dataSummary.R [hunk 24]
```diff
@@ -0,0 +1,655 @@
...
+        color = "aqua",
+        width = 4
+      )
+    })
+    
+    output$order <- shinydashboard::renderInfoBox({
+      shinydashboard::infoBox(
+        "# of Kingdom",
+        nrow(unique(na.omit(dataset(
+
```
Comment (thiloshon) [inline R/mod_dataSummary.R:473]: why is this blank?

File: R/mod_spatial.R [hunk 25]
```diff
@@ -0,0 +1,592 @@
...
+      monthofYear <- as.numeric(
+        strftime(
+          as.Date(
+            dataForBar$Date_collected,
+            na.rm = T
+          ),
+          format = "%m"
+        )
+      )
+      Year_ = as.numeric(
```
Comment (thiloshon) [inline R/mod_spatial.R:145]: Why are you naming this variable as Year_  ? any special reason for this format?

**Review by thiloshon — CHANGES_REQUESTED** [review]

File: DESCRIPTION (hunk 13 above)
Comment (rahulchauhan049) [reply DESCRIPTION:3]: what date should i add?

File: R/mod_spatial.R (hunk 25 above)
Comment (rahulchauhan049) [reply R/mod_spatial.R:145]: Actually, I copied this line from here https://github.com/vijaybarve/bdvis/blob/master/R/tempolar.R#L72
But now I don't need this.

**Pull request closed without merging**